bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/IammeterBaseHandler.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.iammeter.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.cache.ExpiringCache;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IammeterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Yang Bo - Initial contribution
+ */
+
+@NonNullByDefault
+public class IammeterBaseHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IammeterBaseHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private IammeterConfiguration config;
+    private static final int TIMEOUT_MS = 5000;
+    private final ExpiringCache<Boolean> refreshCache = new ExpiringCache<>(Duration.ofSeconds(5), this::refresh);
+
+    public IammeterBaseHandler(Thing thing) {
+        super(thing);
+        config = getConfiguration();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refreshCache.getValue();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        config = getConfiguration();
+        if (refreshJob == null) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
+            this.refreshJob = refreshJob;
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    protected void resolveData(String response) {
```

Comment:
By maing this class abstract you can also make this method abstract:
```suggestion
    protected abstract void resolveData(String response);
```

Reply:
changed.